### PR TITLE
Implement missing CSE endpoints

### PR DESCRIPTION
**Pull request by ccorsin**

### CODE REVIEW

File: crate/server/src/routes/google_cse/operations.rs [hunk 1]
```diff
@@ -312,12 +313,14 @@ pub async fn private_key_sign(
     let database_params = kms.get_sqlite_enc_secrets(&req_http)?;
 
     debug!("private_key_sign: validate_tokens");
-    let user = validate_tokens(
+    let roles: &[&str; 1] = &["signer"];
```
Comment (Manuthor) [inline crate/server/src/routes/google_cse/operations.rs:316]: create a const variable if it is really hardcoded

File: crate/server/src/routes/google_cse/operations.rs [hunk 2]
```diff
@@ -448,73 +471,527 @@ pub async fn private_key_decrypt(
...
+        KmsError::InvalidRequest("Invalid wrapped key - missing data.".to_owned())
+    })?;
+    let iv_counter_nonce = dek.iv_counter_nonce.ok_or_else(|| {
+        KmsError::InvalidRequest("Invalid wrapped key - missing nonce.".to_owned())
+    })?;
+    let authenticated_encryption_tag = dek.authenticated_encryption_tag.ok_or_else(|| {
+        KmsError::InvalidRequest("Invalid wrapped key - authenticated encryption tag.".to_owned())
+    })?;
+
+    let wrapped_dek = [iv_counter_nonce, data, authenticated_encryption_tag].concat();
```
Comment (Manuthor) [inline crate/server/src/routes/google_cse/operations.rs:606]:
```suggestion

    let mut wrapped_dek = Vec::with_capacity(iv_counter_nonce.len() + data.len() + authenticated_encryption_tag.len());
    wrapped_dek.extend_from_slice(&iv_counter_nonce);
    wrapped_dek.extend_from_slice(&data);
    wrapped_dek.extend_from_slice(&authenticated_encryption_tag);
```

I believe `extend_from_slice` does less copies than `concat`.

File: crate/cli/src/actions/google/keypairs/create_keypairs.rs [hunk 3]
```diff
@@ -57,6 +61,10 @@ pub struct CreateKeypairsAction {
         required = true
     )]
     subject_name: String,
+
+    /// The existing private key id of an existing RSA keypair to use (optionnal - if no ID is provided, a RSA keypair will be created)
+    #[clap(long = "user-private-key-id", short = 'k')]
+    rsa_private_key_id: Option<String>,
```
[the remark this answers is not in the document]
Comment (ccorsin) [reply crate/cli/src/actions/google/keypairs/create_keypairs.rs:67]: it's still there, no? :)

File: crate/server/src/routes/google_cse/operations.rs (hunk 1 above)
Comment (ccorsin) [reply crate/server/src/routes/google_cse/operations.rs:316]: I put an enum in the end - is this OK for you ?

File: crate/server/src/routes/google_cse/operations.rs (hunk 2 above)
Comment (ccorsin) [reply crate/server/src/routes/google_cse/operations.rs:606]: I updated all the occurences

File: crate/cli/src/actions/google/keypairs/create_keypairs.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Manuthor) [reply crate/cli/src/actions/google/keypairs/create_keypairs.rs]: OK. So in that case, just remove the "user-private-key-id", thanks!

File: crate/server/src/routes/google_cse/jwt.rs [hunk 4]
```diff
@@ -187,33 +231,29 @@ pub(crate) async fn validate_tokens(
...
-        KmsError::Unauthorized("Authorization token should contain an email".to_owned())
-    })?;
-    kms_ensure!(
-        authorization_email == authentication_email,
-        KmsError::Unauthorized(
-            "Authentication and authorization emails in tokens do not match".to_owned()
-        )
-    );
-
+    #[cfg(any(not(feature = "insecure"), not(test)))]
```
Comment (Manuthor) [inline crate/server/src/routes/google_cse/jwt.rs:234]: brain killer this cfg!

File: crate/cli/src/actions/google/keypairs/create_keypairs.rs [hunk 5]
```diff
@@ -35,15 +39,15 @@ const EXTENSION_CONFIG: &[u8] = b"[ v3_ca ]
...
+    #[clap(required = true)]
     user_id: String,
 
     /// CSE key ID to wrap exported user private key
     #[clap(long = "csekey-id", short = 'w', required = true)]
     csekey_id: String,
 
     /// The issuer certificate id
-    #[clap(long = "issuer_private_key_id", short = 'i')]
+    #[clap(long = "issuer-private-key-id", short = 'i', required = true)]
```
[the remark this answers is not in the document]
Comment (Manuthor) [reply crate/cli/src/actions/google/keypairs/create_keypairs.rs:50]:
```suggestion
    #[clap(long, short = 'i', required = true)]
```

File: crate/cli/src/actions/google/keypairs/create_keypairs.rs [hunk 6]
```diff
@@ -35,15 +39,15 @@ const EXTENSION_CONFIG: &[u8] = b"[ v3_ca ]
 #[clap(verbatim_doc_comment)]
 pub struct CreateKeypairsAction {
     /// The requester's primary email address
-    #[clap(long = "user-id", short = 'u', required = true)]
+    #[clap(required = true)]
```
Comment (Manuthor) [inline crate/cli/src/actions/google/keypairs/create_keypairs.rs:42]:
```suggestion
    #[clap(long, required = true)]
```

**Pull request closed without merging**